src/main/java/com/example/Tictactoe.java

```diff
@@ -0,0 +1,102 @@
+package com.example;
+
+public class Tictactoe {
+
+	public static char getWinner(char[][] board) {
+		validateBoard(board);
+
+		int checkRow1 = 0;
+		int checkRow2 = 0;
+		int checkRow3 = 0;
+		int checkColumn1 = 0;
+		int checkColumn2 = 0;
+		int checkColumn3 = 0;
+		int checkDiagonal1 = 0;
+		int checkDiagonal2 = 0;
+		int rowIndex;
+		int columnIndex;
+
+		//checking rows
+		rowIndex = 0;
+		while (rowIndex < 3) {
+			columnIndex = 0;
+			while (columnIndex < 3) {
+				switch (rowIndex) {
+					case 0 -> checkRow1 = increaseCheck(board[rowIndex][columnIndex], checkRow1);
+					case 1 -> checkRow2 = increaseCheck(board[rowIndex][columnIndex], checkRow2);
+					case 2 -> checkRow3 = increaseCheck(board[rowIndex][columnIndex], checkRow3);
+				}
+				columnIndex++;
+			}
+			rowIndex++;
+		}
+
+		//checking columns
+		columnIndex = 0;
+		while (columnIndex < 3) {
+			rowIndex = 0;
+			while (rowIndex < 3) {
+				switch (columnIndex) {
+					case 0 -> checkColumn1 = increaseCheck(board[rowIndex][columnIndex], checkColumn1);
+					case 1 -> checkColumn2 = increaseCheck(board[rowIndex][columnIndex], checkColumn2);
+					case 2 -> checkColumn3 = increaseCheck(board[rowIndex][columnIndex], checkColumn3);
+				}
+				rowIndex++;
+			}
+			columnIndex++;
+		}
+
+		//checking diagonals
+		columnIndex = 0;
+		while (columnIndex < 3) {
+			checkDiagonal1 = increaseCheck(board[columnIndex][columnIndex], checkDiagonal1);
+			checkDiagonal2 = increaseCheck(board[2 - columnIndex][columnIndex], checkDiagonal2);
+			columnIndex++;
+		}
+
+		//results
+		int flag = checkToFlag(checkRow1) * checkToFlag(checkRow2) * checkToFlag(checkRow3) *
```

Comment:
As I commented on the test case, I don't think this logic is good. In theory the following game can be valid:
```
X X X
O O X
O O X
```